### ISSUE TITLE
Default arguments in signals

### ISSUE THREAD

**Issue by vonagam** [godotengine/godot-headers#88]
There are [`default_args`](https://github.com/godotengine/godot-headers/blob/4bae481fc305639a076a69485bed37679149bd32/nativescript/godot_nativescript.h#L186), [`num_default_args`](https://github.com/godotengine/godot-headers/blob/4bae481fc305639a076a69485bed37679149bd32/nativescript/godot_nativescript.h#L185) in `godot_signal` and [`default_value`](https://github.com/godotengine/godot-headers/blob/4bae481fc305639a076a69485bed37679149bd32/nativescript/godot_nativescript.h#L178) in `godot_signal_argument`.

Based on source code for [`godot_nativescript_register_signal`](https://github.com/godotengine/godot/blob/c7b97f077a0f62d9e280adbddddc901f35e4f40f/modules/gdnative/nativescript/godot_nativescript.cpp#L162) (link to master, there are no changes to relevant part in question since third version) seems like `default_args` are not used. Instead a number of `num_default_args` `default_value`s is [collected](https://github.com/godotengine/godot/blob/c7b97f077a0f62d9e280adbddddc901f35e4f40f/modules/gdnative/nativescript/godot_nativescript.cpp#L187) from the start of args for use as defaults.

So seems like either `default_args` or `default_value` should be removed.

Also it seems wrong that default values are collected from the start and not the end of `args`.

**Comment by ParadoxV5** [comment]
Excuse me for necroposting, but a quick update: While these args are still present on `3.5` and `3.x`, the overhauled `4.0` GDExtension header (#107) does not expose them and whether they’re still present in the source code (I couldn’t search it) is not us header users’ concern.

**Comment by vonagam** [comment]
I don't think it is necroposting as it is one of few open issues here. And as I understand until your PR is merged it is still valid one. After that, I guess, it can be close as I don't see `default_args` in json of your PR.
